uportal-war/src/main/webapp/WEB-INF/flows/entity-selector/selectEntities.jsp

```diff
@@ -139,180 +126,280 @@
-                                                <c:if test="${showCancelButton}">
-                                                    <input class="button btn btn-link" type="submit" value="<spring:message code="${ cancelButtonCode }" text="${ cancelButtonText }"/>" name="_eventId_cancel"/>
-                                                </c:if>
+                                        </div>
+                                    </div>
+                                    <div class="col-sm-6">
+                                        <div class="panel panel-default">
+                                            <div class="panel-body">
+                                                <h5>Group must <strong>EXCLUDE</strong></h5>
+                                                <!-- Data Exclude Node Tree -->
+                                                <div id="${n}dataExcludes" class="demo"></div>
+                                                <hr/>
+                                                <ul id="${n}dataExcludesList">
+                                                </ul>
                                             </div>
-                                        </form>
+                                        </div>
                                     </div>
                                 </div>
-                            </div><!--end: content-->
-                        </div>
-                    </div><!--end:view-single-select-->
-                </c:otherwise>
-            </c:choose>
-        </div><!--end:portlet-content-->
-    </div><!--end:portlet-->
+                            </form>
+                        </div> <!-- end .modal-content div -->
+                        <div class="modal-footer">
+                            <button id="${n}cancelAdHocButton" type="button" class="btn btn-default" data-dismiss="modal">Close <i class="fa fa-times"></i></button>
+                            <button id="${n}saveAdHocButton" type="button" class="btn btn-primary" data-dismiss="modal">Save changes <i class="fa fa-save"></i></button>
+                        </div> <!-- end .modal-footer div -->
+                    </div> <!-- end .modal-content div -->
+                </div> <!-- end .modal-dialog div -->
+            </div><!--end adhocGroupModal -->
+        </div> <!-- end #content div -->
+    </div><!--end portlet-->
+
+    <script type="text/javascript">
+    </script>
+
     <script type="text/javascript">
+        var setGroupDescription = function() {
+            var includes, excludes, description;
+            descInput = $("#groupDesc");
+            includes = $("li", "#${n}dataIncludesList").map(function () { return $(this).text(); }).get();
+            excludes = $("li", "#${n}dataExcludesList").map(function () { return $(this).text(); }).get();
+
+            if (includes.length + excludes.length == 0) {
+                $("#groupDesc").val("Collection of ad hoc groups");
```

Comment:
I guess maybe this should just be blank;  we're not managing container groups with this UI.